src/tensorinfo.cpp

```diff
@@ -261,6 +261,21 @@ const shape_t& TensorInfo::getShape() const {
     return shape;
 }
 
+void TensorInfo::setShape(const shape_t& shape) {
+    this->shape = shape;
+}
+
+std::shared_ptr<TensorInfo> TensorInfo::createCopyWithAdditionalShape(const shape_t& additionalShape) const {
+    if (additionalShape.size() == 0) {
+        return nullptr;
+    }
+    auto copy = std::make_shared<TensorInfo>(*this);
+    shape_t newShape = this->shape;
+    newShape.insert(newShape.begin() + 1, additionalShape.begin(), additionalShape.end());
```

Comment:
It would be clearer if we would have something like this:

shape_t currentShape = tensorInfo.getShape();
shape_t wantedShape = getGatherNodeShape(currentShape);
auto newTensorInfo = tensorInfo.getCopyWithNewShape(wantedShape);

Then you have logic of handling gather nodes separate, and add basic and cohesive new API for TensorInfo.

Reply:
Done